search/engine.go

```diff
@@ -651,24 +675,13 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		}()
 	}
 
-	var resultTypes []string
-	if sortBy == consts.SORT_BY_NEWER_TO_OLDER || sortBy == consts.SORT_BY_OLDER_TO_NEWER {
-		resultTypes = make([]string, 0)
-		for _, str := range consts.ES_SEARCH_RESULT_TYPES {
-			if str != consts.ES_RESULT_TYPE_COLLECTIONS {
-				resultTypes = append(resultTypes, str)
-			}
-		}
-	} else {
-		resultTypes = consts.ES_SEARCH_RESULT_TYPES
-	}
-
 	intents, err := e.AddIntents(&query, preference, consts.INTENTS_SEARCH_COUNT, sortBy)
 	if err != nil {
 		log.Errorf("ESEngine.DoSearch - Error adding intents: %+v", err)
 	}
 
 	// Filter out duplicates of regular results and intents carousel results
+	// Note: Currently we don't have such logic for 'filter grammar' results.
```

Comment:
Don't understand this comment.

Reply:
When we have a lessons carousel we filter out the regular results that are also exist in the carousel . Currently we don't support showing intent carousels for grammar filtered results. So we are also not filtering many appearances of lesson results.

Reply:
Copy paste your explanation into the comment

Reply:
added